### ISSUE TITLE
Panics without explanation when creating render pipeline with >4 fragment color targets

### ISSUE THREAD

**Issue by Keavon** [gfx-rs/wgpu#2108]
**Description**
When creating a render pipeline, it is limited to 4 color attachments. This limit is set by `MAX_COLOR_TARGETS`. WGPU does not provide an error message explaining this, instead it proceeds and another crate (`arrayvec`) ends up panicking with a very unhelpful error message.

**Repro steps**
WGPU version 0.11

```rust
device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
		...
		fragment: Some(wgpu::FragmentState {
			...
			targets: // Provide 5 or more `ColorTargetState`s here
```

In `wgpu-hal-0.11.4/src/vulkan/device.rs:1456`, the following code tries to call `ArrayVec::push()` which panics if the array is full (and the array capacity is set to `MAX_COLOR_TARGETS`):
```rust
compatible_rp_key.colors.push(super::ColorAttachmentKey {
    base: super::AttachmentKey::compatible(
        vk_format,
        vk::ImageLayout::COLOR_ATTACHMENT_OPTIMAL,
    ),
    resolve: None,
});
```

**Expected vs observed behavior**
A size check against `MAX_COLOR_TARGETS` should occur and a helpful error message should be provided explaining the maximum of 4, and perhaps offering suggested workarounds.

**Platform**
Windows

**Comment by Anti-Alias** [comment]
Hit this same error myself today.
I was wondering if there was a plan to either increase this value or have it be queryable in WGPU.
I assume it's set to 4 for portability reasons.

Currently trying to implement deferred rendering with 5+ attachments:
- position (xyz),
- normal (xyz),
- diffuse/albedo (rgba),
- specular (rgb),
- emissive (rgb)

Not sure if this means I'll need to make multiple passes through the same geometry, or if there's an alternate method that doesn't require me to make 5 attachments.
I considered combining the specular and emissive into one target (emissive(rgb) + specular_intensity(a)), but I want to allow for colored specular highlights.
Still kinda a newbie at WGPU/modern graphics pipelines, so bear with me 😃.

**Comment by Keavon** [comment]
@Anti-Alias if you're in the Rust Gamedev server, check this conversation regarding how to more efficiently use the necessary channels in deferred rendering. I had the same problems before I realized many of those channels are actually unnecessary.

https://discord.com/channels/676678179678715904/676685797524766720/902044811492745216

**Comment by jinleili** [comment]
This issue was fixed by #2913
